locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/networking/SharedLocationSecurityGroupCustomizer.java

```diff
@@ -0,0 +1,172 @@
+import com.google.common.collect.Range;
+import com.google.common.collect.RangeSet;
+
+/**
+ * Configures a shared security group on Jclouds locations
+ * <p>
+ * Is based on {@link JcloudsLocationSecurityGroupCustomizer} but can be instantiated
+ * in yaml e.g.
+ * <p>
+ * services:
+ * - type: org.apache.brooklyn.entity.software.base.EmptySoftwareProcess
+ * brooklyn.config:
+ * provisioning.properties:
+ * customizers:
+ * - $brooklyn:object:
+ * type: org.apache.brooklyn.location.jclouds.networking.SharedLocationSecurityGroupCustomizer
+ * object.fields: {locationName: "test", portRanges: ["900-910","915", "22"], cidr: "82.40.153.101/24"}
+ */
+public class SharedLocationSecurityGroupCustomizer extends BasicJcloudsLocationCustomizer {
+
+    private String locationName = null;
+
+    private List<String> portRanges = null;
+
+    private String cidr = null;
+
+    /**
+     * We track inbound ports from the template to open them once we've created the new
+     * security groups
+     */
+    private int[] inboundPorts;
+
+    /**
+     * The location name is appended to the name of the shared SG - use if you need distinct shared SGs within the same location
+     *
+     * @param locationName appended to the name of the SG
+     */
+    public void setLocationName(String locationName) {
+        this.locationName = locationName;
+    }
+
+    /**
+     * Extra ports to be opened on the entities in the SG
+     *
+     * @param portRanges
+     */
+    public void setPortRanges(List<String> portRanges) {
+        this.portRanges = portRanges;
+    }
+
+    /**
+     * Set to restict the range that the ports that are to be opened can be accessed from
+     *
+     * @param cidr
+     */
+    public void setCidr(String cidr) {
+        this.cidr = cidr;
+    }
+
+    @Override
+    public void customize(JcloudsLocation location, ComputeService computeService, TemplateOptions templateOptions) {
+        super.customize(location, computeService, templateOptions);
+        inboundPorts = templateOptions.getInboundPorts();
+    }
+
+    @Override
+    public void customize(JcloudsLocation location, ComputeService computeService, Template template) {
+        super.customize(location, computeService, template);
+
+        final JcloudsLocationSecurityGroupCustomizer instance = getInstance(getSharedGroupId(location));
+
+        if (cidr != null) instance.setSshCidrSupplier(Suppliers.ofInstance(new Cidr(cidr)));
+
+        instance.customize(location, computeService, template);
+    }
+
+    @Override
+    public void customize(JcloudsLocation location, ComputeService computeService, JcloudsMachineLocation machine) {
+        super.customize(location, computeService, machine);
+
+        final JcloudsLocationSecurityGroupCustomizer instance = getInstance(getSharedGroupId(location));
+
+        if (portRanges != null) {
+            RangeSet<Integer> portRanges = BrooklynNetworkUtils.portRulesToRanges(this.portRanges);
```

Comment:
I'd parse these when they are first set, rather than when they are about to be used - fail fast (rather than after someone has waited for their VM to be provisioned).